src/lightning_app/core/app.py

```diff
@@ -352,17 +356,24 @@ def run_once(self):
 
         _process_command_requests(self)
 
+        t0 = time()
+
         try:
             self.check_error_queue()
-            t0 = time()
-            self.root.run()
-            self._last_run_time = time() - t0
+            # Execute the flow only if:
+            # - There are state changes
+            # - It is the first execution of the flow
+            if self._has_updated or not self.first_execution:
```

Comment:
why is there a negation on the "first execution" check? We shouldn't execute if it's not a first execution (and has_updated is False) right?
```
if self._has_updated or self.first_execution
```

Reply:
Switched off. Thanks.